pandas/core/generic.py

```diff
@@ -1038,6 +1038,181 @@ def equals(self, other):
             return False
         return self._data.equals(other._data)
 
+    # -------------------------------------------------------------------------
+    # Label or Level Combination Helpers
+    #
+    # A collection of helper methods for DataFrame/Series operations that
+    # accept a combination of column/index labels and levels.  All such
+    # operations should utilize/extend these methods when possible so that we
+    # have consistent precedence and validation logic throughout the library.
+
+    _shared_docs['_is_level_reference'] = """
+        Test whether a key is a level reference for a given axis.
```

Comment:
so this is getting very complicated with implementations both in Series & DataFrame. you actually don't need any of that and instead you can write these methods generically here. you *already* know how many axes you have, and what the info axes are (IOW the primary selection axis). Its going to be way less code (not to mention you don't need to add the shared docs).

Reply:
also need to guard against anything with ndim > 2 (just raise).